tools/kubernetes/README.md

```diff
@@ -0,0 +1,140 @@
+# Export KFP Pipelines to Kubernetes Manifests
+
+This tool exports all **Kubeflow Pipelines (KFP)** and their **PipelineVersions**  from the KFP database and generates Kubernetes-compatible Pipeline and PipelineVersion YAML manifests. These manifests can then be deployed directly to a Kubernetes cluster.
+
+## What It Does
+
+The script performs the following steps:
+
+1. **Connects to the KFP API server** to retrieve all pipelines and their versions.
+2. **Converts the retrieved objects** into Kubernetes `Pipeline` and `PipelineVersion` resources.
+3. **Generates Kubernetes-compliant names**:
+   - Pipeline names are transformed to conform to [Kubernetes DNS-1123](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/) naming standards.
+   - Pipeline version names are derived by combining the pipeline name and version display name. If the combined name exceeds length limits, it is truncated and a unique index is appended to avoid duplication.
+4. **Writes the output YAML files** into a specified directory—one file per pipeline, including all its versions.
+
+## Installation
+
+This script depends on the following Python packages:
+
+- `kfp`
+- `pyyaml`
+- `requests`
+
+You can install them via pip:
+
+```bash
+pip install kfp pyyaml requests
+```
+## Usage
+
+```bash
+python export_kfp_pipelines.py \
+  --kfp-server-host $KFP_SERVER_HOST \
+  --token $KFP_BEARER_TOKEN \
+  --ca-bundle $CA_BUNDLE \
+  --namespace your-namespace \
+  --batch-size 10 \
+  --output ./exported-pipelines
+  --no-pipeline-name-prefix
```

Comment:
for any optional parameters (such as `ca-bundle`, for example), I would encapsulate the parameter in square brackets to denote they are optional/have defaults